src/extension/background-script/actions/ln/request.ts

```diff
@@ -0,0 +1,83 @@
+import utils from "~/common/lib/utils";
+import { MessageGenericRequest } from "~/types";
+
+import db from "../../db";
+import state from "../../state";
+
+const request = async (
+  message: MessageGenericRequest
+): Promise<{ data: unknown } | { error: string }> => {
+  const connector = await state.getState().getConnector();
+
+  const { origin, args } = message;
+
+  try {
+    if (!connector.requestMethod) {
+      throw new Error(
+        `${message.args.method} is not supported by your account`
+      );
+    }
+
+    const allowance = await db.allowances
+      .where("host")
+      .equalsIgnoreCase(origin.host)
+      .first();
+
+    if (!allowance?.id) {
+      return { error: "Could not find an allowance for this host" };
+    }
+
+    const permission = await db.permissions
+      .where("host")
+      .equalsIgnoreCase(origin.host)
+      .and((p) => p.method === args.method)
+      .first();
+
+    // request method is allowed to be called
+    if (permission && permission.enabled) {
+      const response = await connector.requestMethod(args.method, args.params);
+      return response;
+    } else {
+      const promptResponse = await utils.openPrompt<{
+        enabled: boolean;
+        blocked: boolean;
+      }>({
+        args: {
+          requestPermission: {
+            method: args.method,
+          },
+        },
+        origin,
+        action: "public/webln/confirmRequestPermission",
```

Comment:
Should this be in `webln` at all? 

Would that mean that if we want to use this popup for nostr we would need to duplicate all the code here? (for storing the permission, etc) Does it make sense to have the different actions implement this or could this be handled by the prompt in the background?

Reply:
No strong opinion on the path to the prompt.
I removed "webln" now.

Reply:
> Would that mean that if we want to use this popup for nostr we would need to duplicate all the code here? (for storing the permission, etc) Does it make sense to have the different actions implement this or could this be handled by the prompt in the background?

My initial thinking was that all the checks are done in the prompt.
This was my start in the other PR in which Bumi then pointed me to code where all is handled in the background script [in this comment](https://github.com/getAlby/lightning-browser-extension/pull/1695#issuecomment-1303298272)

Not sure about the future Nostr integration, but if this would be then more or less the same as for the webln request it would require a re-thinking, yes. 
Either extract the handling into another function which can be re-imported again for Nostr. 
Or move the checks to the prompt.
I would suggest discussing this with Bumi again and open a follow-up PR for this.

Reply:
I guess this is exactly the reason why @bumi wanted to have practical examples implemented within this PR as only with that we are learning about how they actually will be used by different consumers and which parts of the permission system make sense to be shared between those. 

>  I would suggest discussing this with Bumi again and open a follow-up PR for this.

👍